### PR TITLE
Handle Execution ID in dotnet test

### PR DESCRIPTION
**Pull request by mariam-abdulla**

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-test/TestApplication.cs [hunk 1]
```diff
@@ -239,6 +240,10 @@ private string BuildHelpArgs(bool isDll)
 
         public void OnHandshakeInfo(HandshakeInfo handshakeInfo)
         {
+            if (handshakeInfo.Properties.TryGetValue(HandshakeInfoPropertyNames.ExecutionId, out string executionId))
+            {
+                _executionIds.Add(executionId);
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestApplication.cs:245]: Should have an event `OnNewExectutionId(id)`?

**Review by MarcoRossignoli — APPROVED** [review]
question, LGTM

File: src/Cli/dotnet/commands/dotnet-test/TestApplication.cs [hunk 2]
```diff
@@ -18,6 +18,7 @@ internal sealed class TestApplication : IDisposable
 
         private NamedPipeServer _pipeConnection;
         private Task _namedPipeConnectionLoop;
+        private HashSet<string> _executionIds = [];
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestApplication.cs:21]: This one could be accessed in parallel right?

**Review by MarcoRossignoli — CHANGES_REQUESTED** [review]
Pls check the concurrency of the non thread safe structures

File: src/Cli/dotnet/commands/dotnet-test/TestApplication.cs (hunk 2 above)
Comment (mariam-abdulla) [reply src/Cli/dotnet/commands/dotnet-test/TestApplication.cs:21]: > The code looks pointless anyway: the only use of `_executionIds` is the `_executionIds.Add(executionId)` call; nothing enumerates the HashSet or calls `Contains`. Is a separate pull request planned that would start reading the execution IDs from `_executionIds`?

Yes, exactly. There will be another PR that makes use of this Hashset.

**Review by MarcoRossignoli — APPROVED** [review]
:shipit:

**Comment by Evangelink** [comment]
Relates to #45927

**Pull request merged**